webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -407,7 +408,7 @@ const AdvancedTextEditor = ({
         if (!showPreview) {
             focusTextbox();
         }
-    }, [showPreview]);
+    }, [showPreview, selectedPostFocussedAt]);
```

Comment:
This is the actual change that will get the focus on the input box again. The other changes in this PR are small improvements to the code suggested by the linter.

Reply:
I am not sure if posts can get somehow selected on the center pane. If so, a change in the selected post may focus the RHS instead of the center pane.

Either way, this is in the `advanced_text_editor` so this is probably moving the cursor first to the center pane, then to the RHS. This is working in that order probably because how components are rendered in react, but can potentially lead to race conditions.

On top of that, the reason for this effect is only to focus when we stop showing the preview. Adding this other behavior can be confusing.